python-threatexchange/threatexchange/fetcher/fetch_state_base.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+Store and look up data from fetch_api
+"""
+
+
+from dataclasses import dataclass
+from enum import Enum
+import typing as t
+
+from threatexchange.signal_type.signal_base import SignalType
+
+
+class SignalOpinionCategory(Enum):
+    """
+    What the opinion on a signal is.
+
+    Some APIs may not support all of these, but each of these should influence
+    what action you might take as a result of matching, otherwise it might
+    make more sense as a tag
+    """
+
+    FALSE_POSITIVE = 0  # Signal generates false positives
+    WORTH_INVESTIGATING = 1  # Indirect indicator
+    TRUE_POSITIVE = 2  # Confirmed meets category
+
+    DISPUTED = 100  # When multiple opinions - has both false positive and others
+
+
+@dataclass
+class SignalOpinion:
+    owner: int
+    category: SignalOpinionCategory
+    category_labels: t.List[str]
+    additional_labels: t.List[str]
+
+
+@dataclass
+class FetchMetadata:
+    """
```

Comment:
Also add the main details of what it's doing, and not just the weird side details